ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/metrics_correlation/MetricsCorrelation.java

```diff
@@ -177,10 +177,17 @@ public MetricsCorrelationOutput execute(Input input) throws ExecuteException {
         waitUntil(() -> {
             if (modelId != null) {
                 MLModelState modelState = getModel(modelId).getModelState();
-                return modelState == MLModelState.DEPLOYED || modelState == MLModelState.PARTIALLY_DEPLOYED;
+                if (modelState == MLModelState.DEPLOYED || modelState == MLModelState.PARTIALLY_DEPLOYED){
+                    log.info("Model deployed: " + modelState);
+                    return true;
+                } else if (modelState == MLModelState.UNDEPLOYED || modelState == MLModelState.DEPLOY_FAILED) {
+                    log.info("Model not deployed: " + modelState);
+                    deployModel(modelId, ActionListener.wrap(deployModelResponse -> modelId = getTask(deployModelResponse.getTaskId()).getModelId(), e -> log.error("Metrics correlation model didn't get deployed to the index successfully", e)));
+                    return false;
+                }
             }
             return false;
-        }, 10, TimeUnit.SECONDS);
+        }, 120, TimeUnit.SECONDS);
```

Comment:
Why do we change it from 10 to 120?

Reply:
10 seconds is not enough to download the model and downloading the artifacts from djl. Depending on the internet speed it can take longer time.